### PR TITLE
Mypy

### PR DESCRIPTION
**Pull request by ThomasSteinbach**
What does this PR change? What problem does it solve?
-----------------------------------------------------

* mainly fixed isort issues
* added missing pytests
* updated usage of the pipeline:service keyword

Was the change discussed in an issue?
------------------------------------------------------------
no

Checklist
---------

- [x] I have read the [Contribution Guidelines](https://github.com/dbsystel/gitlab-ci-python-library/blob/main/CONTRIBUTING.md)
- [x] I have enabled [maintainer edits for this PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
- [x] I have added tests for all changes in this PR
- [x] I have added api documentation for the changes
- [x] I added a changelog entry to the unreleased section with the PR number. [Changelog](https://github.com/dbsystel/gitlab-ci-python-library/blob/main/CHANGELOG.md)
- [x] Linting is done, and linting jobs are successfull
- [x] I'm done, this Pull Request is ready for review

### CODE REVIEW

File: gcip/core/variables.py [hunk 1]
```diff
@@ -23,7 +19,7 @@ class PredefinedVariables():
     https://docs.gitlab.com/ee/ci/variables/predefined_variables.html
     """
 
-    CHAT_CHANNEL: str = EnvProxy("CHAT_CHANNEL")
+    CHAT_CHANNEL: EnvProxy = EnvProxy("CHAT_CHANNEL")
```
Comment (dvonessen) [inline gcip/core/variables.py:22]: This annotation is not useful, if the user wants to know what `CHAT_CHANNEL`s datatype is, he gets EnvProxy. 
Mybe whe can stop checking this class or we can e.g. cast EnvProxy to str with the `typing` module?

File: .gitlab-ci.py [hunk 2]
```diff
@@ -9,6 +9,7 @@
     python.isort(),
     python.flake8(),
     python.pytest(),
+    python.mypy("gcip"),
```
[the remark this answers is not in the document]
Comment (ThomasSteinbach) [reply .gitlab-ci.py:12]: The argument of this function points to the directory within (!) the project folder, that should be checked. This source code folder must not necessarily be named after the project. CI_PROJECT_NAME should currently be `gitlab-ci-python-library` and not `gcip`.

File: gcip/addons/kaniko/jobs.py [hunk 3]
```diff
@@ -1,6 +1,8 @@
 from typing import Dict, List, Union, Optional
 
-from gcip import Job, Image, PredefinedVariables
+from gcip.core.job import Job
```
[the remark this answers is not in the document]
Comment (ThomasSteinbach) [reply gcip/addons/kaniko/jobs.py:3]: mypy can't handle the class mappings made to the gcip module:

```
gcip/addons/kaniko/jobs.py:3: error: Module 'gcip' has no attribute 'Job'
gcip/addons/kaniko/jobs.py:3: error: Module 'gcip' has no attribute 'Image'
gcip/addons/kaniko/jobs.py:3: error: Module 'gcip' has no attribute 'PredefinedVariables'
```

File: .gitlab-ci.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dvonessen) [reply .gitlab-ci.py:12]: Yes I see. Thanks.

File: gcip/addons/kaniko/jobs.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (dvonessen) [reply gcip/addons/kaniko/jobs.py:3]: Mhh, maybe we should read the mypy documentation further more. But for now, it seems ok.

File: gcip/core/variables.py (hunk 1 above)
Comment (dvonessen) [reply gcip/core/variables.py:22]: For now, this is ok. 
In a future release we will change this to be a classproperty.
See: https://stackoverflow.com/questions/128573/using-property-on-classmethods´

**Review by dvonessen — APPROVED** [review]

**Pull request merged**